text/3855-mitigation-enforcement.md

```diff
@@ -0,0 +1,537 @@
+   As far as I can tell, enforcement makes sense for
+   `-Zsanitizer=cfi, -Zsanitizer=memtag, -Zsanitizer=shadow-call-stack`
+   and does not make sense for
+   `-Zsanitizer=address, -Zsanitizer=dataflow, -Zsanitizer=hwaddress, -Zsanitizer=leak, -Zsanitizer=memory, -Zsanitizer=thread` (`-Z sanitizer=address` should probably be a [target modifier])
+4. [`-Z stack-protector`] - stack smashing protection, local-type mitigation
+5. [`-Z ub-checks`]: as far as I can tell, it is not intended as a mitigation,
+   but since it prevents some UB, it might be thought of as one
+
+[`-Z branch-protection`]: https://github.com/rust-lang/rust/issues/113369
+[`-Z cf-protection`]: https://github.com/rust-lang/rust/issues/93754
+[`-Z ehcont-guard`]: https://learn.microsoft.com/en-us/cpp/build/reference/guard-enable-eh-continuation-metadata?view=msvc-170
+[`-Z indirect-branch-cs-prefix`]: https://github.com/rust-lang/rust/issues/116852
+[`-Z no-jump-tables`]: https://github.com/rust-lang/rust/issues/116592
+[`-C jump-tables`]: https://github.com/rust-lang/rust/pull/145974
+[`-Z retpoline`]: https://github.com/rust-lang/rust/issues/116852
+[`-Z sanitizer`]: https://github.com/rust-lang/rust/issues/89653
+[`-Z stack-protector`]: https://github.com/rust-lang/rust/issues/114903
+[`-Z ub-checks`]: https://github.com/rust-lang/rust/issues/123499
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+When you use a mitigation, such as `-C stack-protector=strong`, if one of your
+dependencies does not have that mitigation enabled, compilation will fail.
+
+> Error: your program uses the crate `foo`, that is not protected by
+> `-C stack-protector=strong`.
+>
+> Recompile that crate with the mitigation enabled, or use
+> `-C allow-partial-mitigations=stack-protector` to allow creating an artifact
+> that has the mitigation only partially enabled.
+>
+> It is possible to disable `-C allow-partial-mitigations=stack-protector` via
+> `-C deny-partial-mitigations=stack-protector`.
+
+Other flags that can be mitigations, for example `-C overflow-checks=on`,
+permit partial mitigations by default, but it is possible to make sure
+your dependencies have the same mitigation setting as you by passing
+`-C deny-partial-mitigations=overflow-checks`. That flag can be
+overridden by `-C allow-partial-mitigations=overflow-checks`.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+For every mitigation-like option, the compiler CLI flags determines whether that
+mitigation allows or denies partial mitigations. This can be turned on
+via `-C allow-partial-mitigations=<mitigation>` and turned off by
+`-C deny-partial-mitigations=<mitigation>` (for example,
+`-C allow-partial-mitigations=stack-protector` and
+`-C deny-partial-mitigations=stack-protector`).
+
+These flags act like every other compiler flag, with the last flag winning if there are multiple
+values.
```

Comment:
Can we clarify what this means for specifying different mitigations? Eg, what is the behavior of `-C allow-partial-mitigations=stack-protector -Callow-partial-mitigations=overflow-checks`?